src/virtualenv/seed/embed/via_app_data/pip_install/base.py

```diff
@@ -31,14 +32,10 @@ def _sync(self, src, dst):
 
     def install(self, version_info):
         self._extracted = True
+        self._uninstall_previous_version()
         # sync image
         for filename in self._image_dir.iterdir():
             into = self._creator.purelib / filename.name
-            if into.exists():
```

Comment:
@gaborbernat I'm sorry, this is NOT an equivalent change. 

We went from "delete target prior to copying into" into "uninstall previous distro but do not delete target directory if it wasn't part of the distro". 

I specifically preserved that functionality because if there are extraneous files that are NOT part of a valid distro, venv would override them on install and won't fail. Now VEnv will fail as soon as it encounters a directory/file that exists which can happen if overwriting a partially created VEnv directory.

Reply:
Hmm, how do you end up with a partially created venv? Do you have some real-world examples of this or are you talking HYPOTHETICALS?

Reply:
Ctrl-C during venv installation with partial install of the package. This will occur intermittently and how intermittent it'll be will depend on the PIP package installation logic which will also change from time to time (whether they write dist-info first or they write package data first etc).

My point is: there was specific logic here that I preserved (not introduced!) that resulted in a very specific behavior that made venv more robust. The cleanup resulted in that logic being removed which made venv less robust and more error prone while the problem I tried to fix was fixed.

Can we please not make venv less brittle and reintroduce the logic that was there before?

Reply:
This is virtualenv, not venv :blush: please.

Reply:
This change you introduced relies on everything else (PIP, wheel, bdist_wheel) working 100% correctly 100% of the time: fully installed packages with 100% correct RECORD file that recorded all the files that were actually installed and that package is installed successfully 100% of the time and never interrupted in the process of installation by user action or running out of disk space.

Reply:
Ok.